Workflow/Sources/Debugging.swift

```diff
@@ -148,3 +148,23 @@ extension WorkflowHierarchyDebugSnapshot {
         }
     }
 }
+
+// MARK: - Compatibility
+
+/// These extensions are utilities to support conditionally emitting debug info only when a
+/// `debugger` is set.
+extension WorkflowUpdateDebugInfo? {
+    var unwrappedOrErrorDefault: WorkflowUpdateDebugInfo {
+        self ?? .unexpectedlyMissing
+    }
+}
+
+extension WorkflowUpdateDebugInfo {
+    fileprivate static let unexpectedlyMissing = {
```

Comment:
in a few places we want to propagate debug info with the 'right' optionality. either it should be `nil` if there's no debugger, or non-`nil` if there is. at the root level we need to emit a value to the debugger if present that is non-optional, hence this 'sanity check' value that will hopefully ensure we don't mess things up.

Reply:
Is there an integration test we could write to validate this behavior, or some other mechanism to solidify this contract?

Reply:
i added a few more unit tests to cover the 'set a debugger' and 'didn't set a debugger' cases